src/fmt/mod.rs

```diff
@@ -287,7 +291,55 @@ impl<'a> DefaultFormat<'a> {
     }
 
     fn write_args(&mut self, record: &Record) -> io::Result<()> {
-        writeln!(self.buf, "{}", record.args())
+        match self.indent {
+            
+            // Fast path for no indentation
+            None => writeln!(self.buf, "{}", record.args()),
+
+            Some(indent_count) => {
+
+                // Create a wrapper around the buffer only if we have to actually indent the message
+
+                struct IndentWrapper<'a, 'b: 'a> {
+                    fmt: &'a mut DefaultFormat<'b>,
+                    indent_count: usize
+                }
+
+                impl<'a, 'b> Write for IndentWrapper<'a, 'b>  {
+                    fn write(&mut self, buf: &[u8]) -> io::Result<usize> {
+                        let mut first = true;
+                        for chunk in buf.split(|&x| x == b'\n') {
+                            if !first {
+                                let bar = self.fmt.subtle_style("|");
```

Comment:
Hmm, I think for our fixed-width indentation we could get away without the `|` here. It would also simplify any regex-style text extraction that wants to de-indent the message.

Reply:
Yes, it makes sense, we can also go without the `|`.